bool loc_0
bool loc_1
bool loc_2
bool bridge
bool des